[OPENY-176] Decouple camp and location finder paragraph

anprok opened the pull request:
## Steps for review

- [x] Create new landing page with Location finder filter and location finder paragraph
- [x] make sure that camps displayed 
- [x] visit admin/openy/extend/uninstall
- [x] uninstall Camps.
- [x] make sure that camp uninstalled without errors
- [x] visit page with map
- [x] make sure that camps not displayed

https://github.com/ymcatwincities/openy/issues/1211

anprok committed:
[OPENY-177] Move locations view to openy_map module and remove obsolete displays for branch, camp, facilities

anprok committed:
[OPENY-177] Update openy_map module

anprok committed:
[OPENY-177] Location finder paragraph dependencies clean up

hamrant commented on modules/custom/openy_map/config/install/openy_map.settings.yml:
```diff
@@ -17,8 +17,8 @@ type_labels:
+  # active type for camp will be added when openy_loc_camp module installed
 type_icons:
   branch: '/profiles/contrib/openy/modules/custom/openy_map/img/map_icon_blue.png'
   camp: '/profiles/contrib/openy/modules/custom/openy_map/img/map_icon_green.png'
```
I think we need move all related to camps parts too

hamrant replied:
I mean type_icons and etc

hamrant commented on modules/custom/openy_map/openy_map.install:
```diff
@@ -75,3 +75,24 @@ function openy_map_update_8002() {
 }
+
+/**
+ * Update description for fields.
```
wrong description

hamrant commented on modules/openy_features/openy_location/modules/openy_loc_camp/openy_loc_camp.install:
```diff
@@ -24,6 +24,34 @@ function openy_loc_camp_update_dependencies() {
+  $config_factory = Drupal::configFactory();
+  if (\Drupal::service('module_handler')->moduleExists('openy_map')) {
+    // Add camp as active type to display camps on location finder map.
+    $config_factory
```
For this better to use $config_updater service, otherwise config will be mark as manual edited

hamrant commented on modules/openy_features/openy_location/modules/openy_loc_camp/openy_loc_camp.install:
```diff
@@ -24,6 +24,34 @@ function openy_loc_camp_update_dependencies() {
+  $config_factory = Drupal::configFactory();
+  if (\Drupal::service('module_handler')->moduleExists('openy_map')) {
+    // Remove camp form active type on map.
+    $config_factory
```
For this better to use $config_updater service, otherwise config will be mark as manual edited

gundevel commented:
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build1091 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build1091 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build1091 |
| Installation process | http://install.openy.cibox.tools/build1091/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1091/

gundevel commented:
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build1094 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build1094 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build1094 |
| Installation process | http://install.openy.cibox.tools/build1094/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1094/

ddrozdik commented:
retest this please

bjwilkins commented:
Build comment file: 
--------------

 
| Environment | Link |
| ------------- | ------------- |
| Fresh OpenY installation - Rose theme | http://openy.cibox.tools/build1095 |
| Fresh OpenY installation - Lily theme | http://lily.openy.cibox.tools/build1095 |
| Upgraded(upgrade path) installation | http://upgrade.openy.cibox.tools/build1095 |
| Installation process | http://install.openy.cibox.tools/build1095/install.php |

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/1095/

ddrozdik commented:
@anprok After enabling Camps back they do not appear on the Location finder

anprok commented:
When removing the Camp package, the camp nodes are also deleted. Therefore, after re-enabling the camp package, you need to add camps manually. I have added the camp manually and it is displayed without problems.
![image](https://user-images.githubusercontent.com/16559938/47665296-709b9300-dba9-11e8-944e-85fef7141049.png)
